terraform/modules/facilitator/facilitator.tf

```diff
@@ -0,0 +1,95 @@
+variable "peer_share_processor_name" {
+  type = string
+}
+
+variable "environment" {
+  type = string
+}
+
+variable "gcp_project" {
+  type = string
+}
+
+# This is the role in AWS we use to construct policy on the S3 buckets. It is
+# configured to allow access to the GCP service account for this facilitator via
+# Web Identity Federation
+# https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_oidc.html
+resource "aws_iam_role" "bucket_role" {
+  name = "prio-${var.environment}-${var.peer_share_processor_name}-bucket-role"
+  # We currently use a single role per facilitator to gate read/write access to
+  # all buckets. We could use audiences to define more roles for read/write on
```

Comment:
I think audiences wouldn't be very appropriate for this, they're more of a tool to avoid improper use of the OIDC tokens. The client can set them to whatever they want, so they don't provide any separation of privilege. Instead we'd probably either have separate service accounts (which provide actual privsep by being separate identities), or use [session policies](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies.html#policies_session) if we want to voluntarily downscope permissions.

Reply:
Yes, you're right, I'm not sure why I referred to audiences when I wrote this. We would do something like separate GCP service accounts for the different map steps and have them assume different AWS roles, and then the `sub` value in the policy would correspond to the GCP SA ID. I'll remove the mention of audience in this comment.